python/mxnet/symbol.py

```diff
@@ -1232,61 +1355,78 @@ def grad(self, wrt):
     # pylint: enable= no-member
 
     def eval(self, ctx=cpu(), **kwargs):
-        """Evaluate a symbol given arguments
+        """Evaluates a symbol given arguments.
 
         The `eval` method combines a call to `bind` (which returns an executor)
         with a call to `forward` (executor method).
         For the common use case, where you might repeatedly evaluate with same arguments,
         eval is slow.
         In that case, you should call `bind` once and then repeatedly call forward.
-        Eval allows simpler syntax for less cumbersome introspection.
+        This function allows simpler syntax for less cumbersome introspection.
+
+        Example usage:
+        ----------
+        >>> a = mx.sym.Variable('a')
+        >>> b = mx.sym.Variable('b')
+        >>> c = a + b
+        >>> ex = c.eval(ctx = mx.cpu(), a = mx.nd.ones([2,3]), b = mx.nd.ones([2,3]))
+        >>> ex
+        [<NDArray 2x3 @cpu(0)>]
+        >>> ex[0].asnumpy()
+        array([[ 2.,  2.,  2.],
+               [ 2.,  2.,  2.]], dtype=float32)
 
         Parameters
         ----------
         ctx : Context
             The device context the generated executor to run on.
 
-        kwargs : list of NDArray or dict of str to NDArray
-            Input arguments to the symbol.
-
-            - If type is list of `NDArray`, the position is in the same order of `list_arguments`.
-            - If type is dict of str to `NDArray`, then it maps the name of arguments
-              to the corresponding `NDArray`.
-            - In either case, all the arguments must be provided.
+        kwargs : Keyword arguments of type `NDArray`
+            Input arguments to the symbol. All the arguments must be provided.
 
         Returns
         ----------
-        result :  a list of NDArrays corresponding to the values
-        taken by each symbol when evaluated on given args.
-        When called on a single symbol (not a group),
+        result :  a list of NDArrays corresponding to the values taken by each symbol when
+        evaluated on given args. When called on a single symbol (not a group),
         the result will be a list with one element.
         """
         return self.bind(ctx, kwargs).forward()
 
 
 
 def var(name, attr=None, shape=None, lr_mult=None, wd_mult=None, dtype=None, init=None, **kwargs):
-    """Create a symbolic variable with specified name.
+    """Creates a symbolic variable with specified name.
+    Equivalent to mx.sym.Variable.
+
+    Example usage:
+    ----------
+    >>> data = mx.sym.Variable('data', attr={'a': 'b'})
+    >>> data
+    <Symbol data>
+    >>> data = mx.sym.Variable('data', shape = (2,3), __shape__=(9,0))
```

Comment:
@piiswrong I added shape to show that if it is specified in kwargs(__shape__), it will overwrite shape from (2,3) to (9,0).
Should I remove it?